authorization/v2/common/uniformResourceIdentifier.go

```diff
@@ -296,7 +296,18 @@ func (uri UniformResourceIdentifier) isSubsetOf(target UniformResourceIdentifier
 	return true
 }
 
-// isSubsetOfAtLeastOne checks if the URI is a subset of at least one of the given URIs
+// GetAllSubsetTargets checks if the URI is a subset of the target and returns all those matching target uris
+func (uri UniformResourceIdentifier) GetAllSubsetTargets(targets []UniformResourceIdentifier) []UniformResourceIdentifier {
```

Comment:
Shouldn't this be called `GetAllSupersetTargets`? The `uri` is the subset and the `targets` are the supersets, right?

Also `GetAllSupersets` would probably be a bit cleaner as well.

Reply:
I don't think so, the `targets` in this case are all the URIs that we are seeing if we have access to (e.g. the URIs from the GetAuthorizedResources) and the `uri` is a URI which we have permission to access

Reply:
Why are we checking if `uri` is subset of `targets` here? https://github.com/unicsmcr/hs_auth/pull/102/files/ca5dbdc05c5708d4d258542b9193ecfa07566c65#diff-f65691a1ca4046207a4ac7b0c20af646ea9a2f819ba66e9b02806639aaf46921R303

Is this supposed to be the other way round?

Reply:
Discussed offline, clarified naming of URI comparison functions.